worktree.go

```diff
@@ -227,20 +227,21 @@ func (w *Worktree) createBranch(opts *CheckoutOptions) error {
 }
 
 func (w *Worktree) getCommitFromCheckoutOptions(opts *CheckoutOptions) (plumbing.Hash, error) {
-	if !opts.Hash.IsZero() {
-		return opts.Hash, nil
-	}
+	hash := opts.Hash
+	if hash.IsZero() {
+		b, err := w.r.Reference(opts.Branch, true)
+		if err != nil {
+			return plumbing.ZeroHash, err
+		}
 
-	b, err := w.r.Reference(opts.Branch, true)
-	if err != nil {
-		return plumbing.ZeroHash, err
-	}
+		if !b.Name().IsTag() {
+			return b.Hash(), nil
+		}
 
-	if !b.Name().IsTag() {
-		return b.Hash(), nil
+		hash = b.Hash()
 	}
 
-	o, err := w.r.Object(plumbing.AnyObject, b.Hash())
+	o, err := w.r.Object(plumbing.AnyObject, hash)
```

Comment:
Keeping `plumbing.AnyObject` here enables an annotated tag to point to another. I am not sure this (nested tags) is supported upstream.

Reply:
You can checkout an annotated tag commit hash upstream `git checkout <tag hash>`